### DISCUSSION TITLE
Make the KedroContext available in the after_catalog_created hook

### DISCUSSION THREAD

**Discussion by antoinebon** [kedro-org/kedro#1576] [Idea]
I'm writing a hook implementation that will dynamically add some datasets in the catalog once `after_catalog_created`. I need to have access to the KedroContext in the hook implementation in order to properly form absolute paths with the `context._project_path`, however this is currently not available in that particular hook.

I can go around this by adding a dummy dataset in the catalog.yaml and reverse engineer the project_path, however extending the hook to provide the context would be much cleaner. Do you think it makes sense?

**Comment by datajoely** [comment]
So I don't think we'll do this - the context is being re-designed and may even cease to exist in the future.

That being said you should be able to do the following which is basically what happens in the context anway:

```python
from pathlib import Path
from  kedro.framework.startup import bootstrap_project
project_path = Path.cwd()
metadata = bootstrap_project(project_path)
```

**Reply by antoinebon** [reply]
OK great, thank you very much for your prompt feedback!
Also the snippet you gave me works like a charm, so thanks once more!
Looking forward to the re-design, when is it planned for?

**Reply by antoinebon** [reply]
Actually the snippet fails in the notebook environment:

```RuntimeError: Could not find the project configuration file 'pyproject.toml' in /code/kedroproject/notebooks. If you have created your project with Kedro version <0.17.0, make sure to update your project template. See https://github.com/kedro-org/kedro/blob/main/RELEASE.md#migration-guide-from-kedro-016-to-kedro-0170 for how to migrate your Kedro project.```

**Reply by datajoely** [reply]
So in notebook world you just need to do `Path.cwd().parent`

**Comment by antonymilne** [comment]
Since kedro 0.18.1 there's an exciting new hook, `after_context_created`, which enables you to do this much more easily 🙂 
```
class Hooks:
    @hook_impl
    def after_context_created(self, context):
        self.project_path = context.project_path

    @hook_impl
    def after_catalog_created(self, catalog):
        catalog.add(...) # use self.project_path here

```
